pycheribuild/config/compilation_targets.py

```diff
@@ -276,6 +277,22 @@ def triple_for_target(cls, target: "CrossCompileTarget", config: "CheriConfig",
         # TODO: do we need any special cases here?
         return target.cpu_architecture.value + common_suffix
 
+    @property
+    def freebsd_target_arch_dir(self):
+        """
+        Return the name of the target architecture machine-dependent code directory
+        in the source tree. (e.g. <arch> in sys/<arch>)
+        """
```

Comment:
Isn't this TARGET_CPUTYPE, i.e. we should call this function freebsd_target_cputype?

Reply:
Uh, TARGET_CPUARCH even (TARGET_CPUTYPE is like -mcpu).

Reply:
This is what cheribsd.py's arch_build_flags does, too, so if you fix this to be `freebsd_target` then you can remove most of that `if` and leave behind just the CHERI handling.